hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestVirtualHostStyleFilter.java

```diff
@@ -106,7 +107,8 @@ public void testVirtualHostStyle() throws  Exception {
             ".localhost:9878", "/myfile", null, true);
     virtualHostStyleFilter.filter(containerRequest);
     URI expected = new URI("http://" + s3HttpAddr +
-        "/mybucket/myfile");
+        "/mybucket/myfile?" + S3Consts.ENDPOINT_STYLE_PARAM + "=" +
+        S3Consts.ENDPOINT_STYLE_VIRTUAL);
```

Comment:
According to examples in [s3 docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html), no extra query parameters are required. The tests should run the same way.

Reply:
Thanks for the review @errose28 I added the extra parameters are added internally in VirtualHostStyleFilter.java so that when we do signature validation late we know to use the original request URI sent by the client and not the URI with bucket name that is appended in VirtualHostStyleFilter.java.